pkg/launcher/launcher.go

```diff
@@ -12,72 +12,61 @@ type ClusterLauncher struct {
 }
 
 type launcherSettings struct {
-	collapseLoginCommand bool
+	// Future Usage Possibly
 }
 
-func NewClusterLauncher(terminal []string, clusterLoginCommand []string) (ClusterLauncher, error) {
+func NewClusterLauncher(terminal string, clusterLoginCommand string) (ClusterLauncher, error) {
+
 	launcher := ClusterLauncher{
-		terminal:            terminal,
-		clusterLoginCommand: clusterLoginCommand,
+		terminal:            strings.Split(terminal, " "),
+		clusterLoginCommand: strings.Split(clusterLoginCommand, " "),
 		settings:            launcherSettings{},
 	}
 
-	return launcher, nil
-}
+	err := launcher.validate()
+	if err != nil {
+		return ClusterLauncher{}, err
+	}
 
-func (l *ClusterLauncher) SetCollapseLoginCommand(setting bool) {
-	l.settings.collapseLoginCommand = setting
+	return launcher, nil
 }
 
-func (l *ClusterLauncher) Validate() error {
+func (l *ClusterLauncher) validate() error {
```

Comment:
I had originally put the validation logic inside the initializer, but then decided to move it out in the case where someone may not want to configure the auto login and just use the other features. I guess when I think about it more then the defaults would take over so it's not like it would fail init 🤔

Reply:
Ah, yeah, that's a fair point.  I hadn't even considered someone would not want to use that feature, because it's such a big part of my workflow. Let me do some thinking this weekend.

Reply:
After thinking about this, I think it probably makes more sense to just have un-configured features behave as defaults, or just not do anything.

Probably more key is to make sure errors from the spawned process write back, even if it's just "process exited with error code X" in some way that's consumable by SREPD but not too blocking. 

The current "critical error" handler is not refined at all, and just there to catch those cases, but could be updated to display in a better way and provide more useful feedback without a full-window BSOD-style message.